phpseclib/Crypt/Random.php

```diff
@@ -54,66 +54,21 @@ class Random
      */
     static function string($length)
     {
-        if (version_compare(PHP_VERSION, '7.0.0', '>=')) {
-            try {
-                return random_bytes($length);
-            } catch (\Error $e) {
-                // If a sufficient source of randomness is unavailable, random_bytes() will emit a warning.
-                // We don't actually need to do anything here. The string() method should just continue
-                // as normal. Note, however, that if we don't have a sufficient source of randomness for
-                // random_bytes(), most of the other calls here will fail too, so we'll end up using
-                // the PHP implementation.
-            }
+        // paragonie/random_compat defines random_bytes() for us if we're using
+        // PHP < 7.0.0. If it can't find any good sources of randomness,
+        // random_bytes() is defined as a function that just throws an exception.
+        try {
+            return \random_bytes($length);
+        } catch (\Throwable $e) {
```

Comment:
See https://github.com/paragonie/random_compat/pull/56 you are better off catching `\Error` and `\Exception` that way you get the same execution path on PHP 5 and PHP 7.